components/calendar-list/CalendarList.tsx

```diff
@@ -0,0 +1,51 @@
+import { CardGrid } from '@/components'
+import { COLLECTIONS, DICTIONARY, SITE } from '@/constants'
+import { getPageDataBySlug } from '@/lib'
+import { formatDate } from '@/utils'
+import { FC } from 'react'
+
+export const CalendarList: FC<CalendarListProps> = async (props) => {
+  const result = await getPageDataBySlug(COLLECTIONS.CALENDARS, props.locale, 'date')
+
+  if (result.status === 'error') return <>{result.message}</>
+
+  const content = result.data as CardGridItemType[]
+  const items = content.map((item: CardGridItemType): CardGridItemType => {
```

Comment:
You could pass the `CardGridItemType` to the map function to enforce the time it returns, like:
`const items = content.map<CardGridItemType>((item) => {`

It has the same effect but is cleaner.